### PR TITLE
Use maximum allowed response size for request/response protocols

### PR DESCRIPTION
**Pull request by AndreiEres**
# Description

Adjust the PoV response size to the default values used in the substrate.
Fixes https://github.com/paritytech/polkadot-sdk/issues/5503

## Integration

The changes shouldn't impact downstream projects since we are only increasing the limit.

## Review Notes

You can't see it from the changes, but it affects all protocols that use the `POV_RESPONSE_SIZE` constant.
- Protocol::ChunkFetchingV1
- Protocol::ChunkFetchingV2
- Protocol::CollationFetchingV1
- Protocol::CollationFetchingV2
- Protocol::PoVFetchingV1
- Protocol::AvailableDataFetchingV1

## Increasing timeouts

https://github.com/paritytech/polkadot-sdk/blob/fae15379cba0c876aa16c77e11809c83d1db8f5c/polkadot/node/network/protocol/src/request_response/mod.rs#L126-L129

I assume the current PoV request timeout is set to 1.2s to handle 5 consecutive requests during a 6s block. This setting does not relate to the PoV response size. I see no reason to change the current timeouts after adjusting the response size.

However, we should consider networking speed limitations if we want to increase the maximum PoV size to 10 MB. With the number of parallel requests set to 10, validators will need the following networking speeds:  
- 5 MB PoV: at least 42 MB/s, ideally 50 MB/s.  
- 10 MB PoV: at least 84 MB/s, ideally 100 MB/s.

The current required speed of 50 MB/s aligns with the 62.5 MB/s specified [in the reference hardware requirements](https://wiki.polkadot.network/docs/maintain-guides-how-to-validate-polkadot#reference-hardware). Increasing the PoV size to 10 MB may require a higher networking speed.

### CODE REVIEW

File: polkadot/node/network/protocol/src/request_response/mod.rs [hunk 1]
```diff
@@ -159,11 +159,8 @@ pub const MAX_PARALLEL_ATTESTED_CANDIDATE_REQUESTS: u32 = 5;
 
 /// Response size limit for responses of POV like data.
 ///
-/// This is larger than `MAX_POV_SIZE` to account for protocol overhead and for additional data in
-/// `CollationFetchingV1` or `AvailableDataFetchingV1` for example. We try to err on larger limits
-/// here as a too large limit only allows an attacker to waste our bandwidth some more, a too low
-/// limit might have more severe effects.
-const POV_RESPONSE_SIZE: u64 = MAX_POV_SIZE as u64 + 10_000;
+/// Same as what we use in substrate networking.
+const POV_RESPONSE_SIZE: u64 = 16 * 1024 * 1024;
```
Comment (sandreim) [inline polkadot/node/network/protocol/src/request_response/mod.rs:163]: I think substrate should export a constant and we should use it here.

**Review by sandreim — APPROVED** [review]
PRDoc ?

File: polkadot/node/network/protocol/src/request_response/mod.rs [hunk 2]
```diff
@@ -159,11 +159,8 @@ pub const MAX_PARALLEL_ATTESTED_CANDIDATE_REQUESTS: u32 = 5;
 
 /// Response size limit for responses of POV like data.
 ///
-/// This is larger than `MAX_POV_SIZE` to account for protocol overhead and for additional data in
-/// `CollationFetchingV1` or `AvailableDataFetchingV1` for example. We try to err on larger limits
-/// here as a too large limit only allows an attacker to waste our bandwidth some more, a too low
-/// limit might have more severe effects.
-const POV_RESPONSE_SIZE: u64 = MAX_POV_SIZE as u64 + 10_000;
+/// Same as what we use in substrate networking.
+const POV_RESPONSE_SIZE: u64 = MAX_RESPONSE_SIZE;
```
Comment (dmitry-markin) [inline polkadot/node/network/protocol/src/request_response/mod.rs:163]: Is it safe to not base this value on `MAX_POV_SIZE`?

**Review by dmitry-markin — APPROVED** [review]

File: polkadot/node/network/protocol/src/request_response/mod.rs (hunk 2 above)
Comment (AndreiEres) [reply polkadot/node/network/protocol/src/request_response/mod.rs:163]: It's okay, I suppose, because we make these changes to bump it.

**Comment by sandreim** [comment]
> > The current required speed of 50 MB/s aligns with the 62.5 MB/s specified [in the reference hardware requirements](https://wiki.polkadot.network/docs/maintain-guides-how-to-validate-polkadot#reference-hardware). Increasing the PoV size to 10 MB may require a higher networking speed.
> 
> Yes, this is worst case scenario, when all blocks you need to recover are full. We'll likely have to raise specs for networking,

Did a bit of math in https://github.com/paritytech/polkadot-sdk/issues/5334#issuecomment-2361120502 . looks like we should be fine.

**Comment by eskimor** [comment]
> I assume the current PoV request timeout is set to 1.2s to handle 5 consecutive requests during a 6s block. This setting does not relate to the PoV response size. I see no reason to change the current timeouts after adjusting the response size.

This is not correct. It is set to that value because of synchronous backing. There we have a very small overall time budget. This is no longer true for asynchronous backing.

Considering that this is a hard timeout where we completely drop the response if exceeded, it might make sense to think about bumping it a bit. This would add robustness, especially with concurrent requests. @AndreiEres

**Pull request merged**